lib/yang.c

```diff
@@ -245,17 +245,25 @@ int yang_snodes_iterate(const struct lys_module *module, yang_iterate_cb cb,
 			if (ret == YANG_ITER_STOP)
 				return ret;
 		}
-		LY_LIST_FOR (&module_iter->compiled->rpcs->node, snode) {
-			ret = yang_snodes_iterate_subtree(snode, module, cb,
-							  flags, arg);
-			if (ret == YANG_ITER_STOP)
-				return ret;
+		if (module_iter->compiled->rpcs) {
+			LY_LIST_FOR (&module_iter->compiled->rpcs->node, snode) {
+				ret = yang_snodes_iterate_subtree(snode, module,
```

Comment:
Which error is returned for this? Basically this idiom is the C version of inherited types. That is, `&rpcs->node` where `node` is an embedded structure (`struct lysc_node`) at the start of `struct lysc_node action`. This allows the elimination of unsafe casting `(struct lysc_node *)rpc` and instead using type-safe checking that an `struct lysc_action_node *` is also a `struct lysc_node *` so `typeof(&rpc->node) == struct lysc_node *)` so `&rpc->node` requires no cast when trying to use as an `struct lysc_node *)`.

In any case what `&foo->bar` translates to is the following:
```
struct foo *foo = 0;
char *result;

result = (char *)foo; /* so result is 0 */
result += offsetof(struct foo, bar);
```

So there is no NULL ptr dereference.

Reply:
the commit message clearly spelled out the message.  The sanitizer is claiming that module_iter->compiled->rpcs is NULL thus derefencing ->node is going to cause fun.